File: python/83_Remove_Duplicates_from_Sorted_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def deleteDuplicates(self, head):
        if not head or not head.__next__:
            return head
        cur=head
        last=head.val
        while cur.__next__ is not None:
            if cur.next.val!=last:
                last=cur.next.val
                cur=cur.__next__
            else:
                cur.next=cur.next.__next__
        return head

File: Java/83_Remove_Duplicates_from_Sorted_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null||head.next==null)return head;
        ListNode slow = head, runner = head;
        while(runner!=null){
            runner = runner.next;
            while(runner!=null&&runner.val==slow.val){
                runner =runner.next;
            }
            slow.next=runner;
            slow=runner;
        }
        return head;
    }
}